app.js

```diff
@@ -1,14 +1,52 @@
 const http = require('http');
+const fs = require('fs');
 
 const hostname = '127.0.0.1';
 const port = 3000;
 
-const server = http.createServer((req, res) => {
-  res.statusCode = 200;
-  res.setHeader('Content-Type', 'text/plain');
-  res.end('Hello World');
+
+// using a read stream. Don't fully understand this
+const htmlPipeFileServer = http.createServer((req, res) => {
+    res.statusCode = 200;
+    res.setHeader('Content-Type', 'text/html');
+    fs.createReadStream('index.html').pipe(res)
+});
+
+
+// application json. You have to pass the json as a string, which is weird.
+const jsonServer = http.createServer((req, res) => {
+    res.statusCode = 200;
+    res.setHeader('Content-Type', 'application/json');
```

Comment:
application json. You have to pass the json as a string, which is weird.

Reply:
Yup. It's sent as plain text. But the header above tells the browser to read it as json.